teps/0021-results-api.md

```diff
@@ -0,0 +1,772 @@
+---
+title: Tekton Results API
+authors:
+  - "@wlynch"
+creation-date: 2020-09-23
+last-updated: 2020-10-26
+status: proposed
```

Comment:
after the discussion in today's API working group, i want to float the idea that we merge this as proposed very quickly (without everything from "proposal" down)

(np if that doesn't help and want to continue the discussion as is)